File: Source/MKL/examples/vmlf/source/vsacosh.f
	programm MKL_VML_TEST

	include "func_interv.fi"

	real*4 sA(VEC_LEN)
	real*4 sB1(VEC_LEN)
	integer i

	MaxRMS=0.0
	do i=1,VEC_LEN
		sA(i)=VML_SACOSH_BEG+((VML_SACOSH_END-VML_SACOSH_BEG)*i)/VEC_LEN
		sB1(i)=0;
	end do

	call VSACOSH(VEC_LEN,sA,sB1)

	print *,"vsAcosh test/example programm"
	print *,""
	print *,""
	print *,"           Argument                     vsAcosh"
	print *,"=============================================================================="
	do i=1,VEC_LEN
		print 10,sA(i),sB1(i)
	end do

10	format(F25.14,F25.14)

	end

File: Source/MKL/examples/vmlc/source/vsacosh.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int main()
{
  float fA[VEC_LEN],fB1[VEC_LEN];

  int i=0,vec_len=VEC_LEN;

  for(i=0;i<vec_len;i++) {
    fA[i]=(float)(__VML_SACOSH_BEG+((__VML_SACOSH_END-__VML_SACOSH_BEG)*i)/vec_len);
    fB1[i]=0;
  }

#if defined(MKL_VML_MIXED)
  vsAcosh(vec_len,fA,fB1);
#elif defined(MKL_VML_LOWER)
  vsacosh(&vec_len,fA,fB1);
#elif defined(MKL_VML_UPPER)
  VSACOSH(&vec_len,fA,fB1);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif


  printf("vsAcosh test/example programm\n\n");
  printf("           Argument                     vsAcosh\n");
  printf("===============================================================================\n");
  for(i=0;i<vec_len;i++) {
    printf("% 25.14f % 25.14f\n",fA[i],fB1[i]);
  }
}
